### DISCUSSION TITLE
Server Error Disconnect, using Streaming API ... Help!

### DISCUSSION THREAD

**Discussion by kiann00** [ig-python/ig-markets-api-python-library#281] [Q&A]
hi there, everyone...... crawling back to seek more help ... 😅

I want to save/download historical data from IG to do some future analysis (as the 10k limit is ridiculous),
I used the main code I found in sample here <https://github.com/ig-python/trading-ig/blob/master/sample/stream_ig.py>, but I am getting the following error
Not sure what I am doing wrong. I show my entire code below.... 🙏

WARNING:trading_ig.lightstreamer:Server error
DISCONNECTED FROM LIGHTSTREAMER

```

#######################
# let's investigate the streaming data service so as to save data for historical analysis

from trading_ig import IGService, IGStreamService
from trading_ig.config import config
from trading_ig.lightstreamer import Subscription

import logging

def on_prices_update(item_update):
    # print("price: %s " % item_update)
    print(
        "{stock_name:<19}: Time {UPDATE_TIME:<8} - "
        "Bid {BID:>5} - Ask {OFFER:>5}".format(
            stock_name=item_update["name"], **item_update["values"]
        )
    )
    
    return

def on_account_update(balance_update):
    print("balance: %s " % balance_update)
    return

ig_service = IGService(user_, password_, api_key_, acc_type_, account_, retryer = retryer)
ig_stream_service = IGStreamService(ig_service)
ig_stream_service.create_session()

# Making a new Subscription in MERGE mode
list_ = ["L1:CS.D.GBPUSD.TODAY.IP", "L1:IX.D.FTSE.DAILY.IP", "IX.D.SPTRD.DAILY.IP"]
subscription_prices = Subscription(mode="MERGE",
        #items=["L1:CS.D.GBPUSD.CFD.IP", "L1:CS.D.USDJPY.CFD.IP"], # sample CFD epics
        items= list_, # sample spreadbet epics
        fields=["UPDATE_TIME", "BID", "OFFER", "CHANGE", "MARKET_STATE", "VOLUME"],
    )


# Adding the "on_price_update" function to Subscription
subscription_prices.addlistener(on_prices_update)
sub_key_prices = ig_stream_service.ls_client.subscribe(subscription_prices)

ig_stream_service.disconnect()


```

**Comment by starsnpixel** [comment]
There must be more than just `WARNING:trading_ig.lightstreamer:Server error DISCONNECTED FROM LIGHTSTREAMER`, right? Or is this really everything you get?

**Comment by LordNadimi** [comment]
Sample is wrong because IG changed something on their side.

Add MARKET: to each item and it will work.

The lightstreamer error you will get each time you include a ig_stream_service.disconnect()

subscription_prices = Subscription(
mode="MERGE",
items=["MARKET:L1:CS.D.GBPUSD.CFD.IP", "MARKET:L1:CS.D.USDJPY.CFD.IP"], # sample CFD epics
items=["MARKET:IX.D.SPTRD.FWM1.IP"]

**Comment by bug-or-feature** [comment]
@kiann00 I can see you removed this code from your script
```
input(
        "{0:-^80}\n".format(
            "HIT CR TO UNSUBSCRIBE AND DISCONNECT FROM \
    LIGHTSTREAMER"
        )
    )
```

Try adding it back in, it's there for a reason
